api/src/main/java/org/openmrs/module/pihmalawi/sql/MysqlScriptRunner.java

```diff
@@ -0,0 +1,207 @@
+    private final Connection connection;
+
+    private final boolean stopOnError;
+    private final boolean autoCommit;
+
+    private String delimiter = DEFAULT_DELIMITER;
+    private boolean fullLineDelimiter = false;
+
+    public MysqlScriptRunner(Connection connection, boolean stopOnError, boolean autoCommit) {
+        this.connection = connection;
+        this.stopOnError = stopOnError;
+        this.autoCommit = autoCommit;
+    }
+
+    public void setDelimiter(String delimiter, boolean fullLineDelimiter) {
+        this.delimiter = delimiter;
+        this.fullLineDelimiter = fullLineDelimiter;
+    }
+    private String getDelimiter() {
+        return delimiter;
+    }
+    /**
+     * Runs an SQL script (read in using the Reader parameter)
+     *
+     * @param reader - the source of the script
+     */
+    public MysqlResult runScript(Reader reader) throws IOException, SQLException {
+        try {
+            boolean originalAutoCommit = connection.getAutoCommit();
+            try {
+                if (originalAutoCommit != this.autoCommit) {
+                    connection.setAutoCommit(this.autoCommit);
+                }
+                return runScript(connection, reader);
+            } finally {
+                connection.setAutoCommit(originalAutoCommit);
+            }
+        } catch (IOException e) {
+            throw e;
+        }  catch (SQLException e) {
+            throw e;
+        }catch (Exception e) {
+            throw new RuntimeException("Error running script.  Cause: " + e, e);
+        }
+    }
+
+    /**
+     * Runs an SQL script (read in using the Reader parameter) using the
+     * connection passed in
+     *
+     * @param conn - the connection to use for the script
+     * @param reader - the source of the script
+     * @throws SQLException if any SQL errors occur
+     * @throws IOException if there is an error reading from the Reader
+     */
+    private MysqlResult runScript(Connection conn, Reader reader) throws IOException,
+            SQLException {
+        MysqlResult result = null;
+        StringBuffer command = null;
+        try {
+            LineNumberReader lineReader = new LineNumberReader(reader);
+            String line;
+            while ((line = lineReader.readLine()) != null) {
+                if (command == null) {
+                    command = new StringBuffer();
+                }
+                String trimmedLine = line.trim();
+                final Matcher delimMatch = delimP.matcher(trimmedLine);
+                if (trimmedLine.length() < 1
+                        || trimmedLine.startsWith("//")) {
+                    // Do nothing
+                } else if (delimMatch.matches()) {
+                    setDelimiter(delimMatch.group(2), false);
+                } else if (trimmedLine.startsWith("--")) {
+                    // Do nothing
+                } else if (trimmedLine.length() < 1
```

Comment:
If we want to support in-line comments, we should add that in here.  These can be useful - I see you had to remove some from an existing script that we have as they were not supported, but we should be in control of that...